solidity/dashboard/src/connectors/read-only-connector.js

```diff
@@ -0,0 +1,40 @@
+import { AbstractConnector } from "./abstract-connector"
+import { WALLETS } from "../constants/constants"
+import { getWsUrl } from "./utils"
+import Web3ProviderEngine from "web3-provider-engine"
+import WebsocketSubprovider from "web3-provider-engine/subproviders/websocket"
+import ExplorerModeSubprovider from "./explorerModeSubprovider";
+import { EventEmitter } from "events"
+
+export class ReadOnlyConnector extends AbstractConnector {
+  /** @type {string} To store selected account by user. */
+  selectedAccount = ""
+
+  constructor() {
+    super(WALLETS.READ_ONLY_ADDRESS.name)
+    this.websocketSubprovider = new WebsocketSubprovider({ rpcUrl: getWsUrl() })
```

Comment:
I think we should create `WebsocketSubprovider` in the `enable` method. Otherwise, it creates an unnecessary connection to the node after creating the `ReadOnlyConnector` object.